pandas/_libs/reduction.pyx

```diff
@@ -529,16 +529,20 @@ def apply_frame_axis0(object frame, object f, object names,
             item_cache.clear()  # ugh
 
             object.__setattr__(slider.dummy, 'name', names[i])
-            piece = f(slider.dummy)
+
+            # issue #21609. Skipping the first group to avoid running
+            # the functions twice. For plots this leads to an extra
+            # plot of the first group.
+            if (i > 0) | (hasattr(piece, 'index')):
```

Comment:
Catching mutation status of the first group turned out to be a headache. The way I have written this right now it only avoids double-runs of first group if function `f` does not output a dataframes/etc (which have index attribute), which is the case for plots.